zookeeper/src/main/java/com/linecorp/armeria/common/zookeeper/DefaultNodeValueCodec.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.zookeeper;
+
+import static java.util.Objects.requireNonNull;
+
+import java.nio.charset.StandardCharsets;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.client.endpoint.EndpointGroupException;
+
+/**
+ * Default {@link NodeValueCodec} implementation which assumes zNode value is a comma-separated
+ * string. Each element of the zNode value represents an endpoint whose format is
+ * {@code <host>[:<port_number>[:weight]]}, such as:
+ * <ul>
+ *   <li>{@code "foo.com"} - default port number, default weight (1000)</li>
+ *   <li>{@code "bar.com:8080} - port number 8080, default weight (1000)</li>
+ *   <li>{@code "10.0.2.15:0:500} - default port number, weight 500</li>
+ *   <li>{@code "192.168.1.2:8443:700} - port number 8443, weight 700</li>
+ * </ul>
+ * the segment and field delimiter can be specified, default will be "," and ":"
+ * Note that the port number must be specified when you want to specify the weight.
+ */
+public class DefaultNodeValueCodec implements NodeValueCodec {
+
+    private final String segmentDelimiter;
+    private final String fieldDelimiter;
+
+    /**
+     * Create a default codec with segment delimiter ',' and field delimiter ':'.
+     */
+    public DefaultNodeValueCodec() {
```

Comment:
We could make this class package local and add a static field to `NodeValueCodec`:

```java
public interface NodeValueCodec {
    /**
     * ... Move the class Javadoc of DefaultNodeValueCodec here ...
     */
    static NodeVauleCodec DEFAULT = DefaultNodeValueCodec.INSTANCE;
    ...
}

final class DefaultNodeValueCodec implements NodeValueCodec {
    static final DefaultNodeValueCodec INSTANCE = new DefaultNodeValueCodec();
    private DefaultNodeValueCodec() { ... }
    ...
}
```